src/libs/actions/Policy.js

```diff
@@ -336,21 +399,31 @@ function invite(logins, welcomeNote, policyID) {
             // Save the personalDetails for the invited user in Onyx
             if (data.jsonCode === 200) {
                 Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, PersonalDetails.formatPersonalDetails(data.personalDetails));
-                Navigation.goBack();
+                const nextPolicy = _.clone(allPolicies[key]);
+                const nextEmployeeList = _.intersection(newEmployeeList, nextPolicy.pendingInvitations);
+                nextPolicy.employeeList = _.uniq([...nextPolicy.employeeList, ...nextEmployeeList]);
+                nextPolicy.pendingInvitations = _.difference(nextPolicy.pendingInvitations, nextEmployeeList);
+                Onyx.set(key, nextPolicy);
+
                 return;
             }
```

Comment:
Let's rever this condition and throw an error with the received error message and then catch it in the catch callback and handle it there.  Then `revertPendingInvitationsList` function will be removed and all of its content will go to `.catch`